### PR TITLE
fix config logic

### PR DESCRIPTION
**Pull request by skylarbarrera**
## What changed (plus any additional context for devs)

config was fine but it assumed users would only have mainnet set but once we introduced testnets that was not the case. 

we still want to update the RPC's if they've changes but now we do it based on the current network vs defaulting to mainnet

## PoW (screenshots / screen recordings)

before: https://cloud.skylarbarrera.com/Screen-Shot-2022-05-26-11-47-16.72.png
https://cloud.skylarbarrera.com/Screen-Shot-2022-05-26-11-47-42.49.png

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?
- [x] Added e2e tests? if not please specify why
- [x] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

File: src/model/config.ts [hunk 1]
```diff
@@ -107,9 +108,11 @@ const init = async () => {
     Logger.sentry('using default config instead...');
   } finally {
     Logger.debug('CURRENT CONFIG', JSON.stringify(config, null, 2));
-    // SET THE DEFAULT PROVIDER AFTER LOADING THE CONFIG
+    // UPDATE THE PROVIDER AFTER LOADING THE NEW CONFIG
+    const currentNetwork = await getNetwork();
     setRpcEndpoints(config);
-    web3SetHttpProvider(networkTypes.mainnet);
+    web3SetHttpProvider(currentNetwork);
+    await saveNetwork(currentNetwork);
```
Comment (estebanmino) [inline src/model/config.ts:115]: do we need to `await` here?

File: src/model/config.ts [hunk 2]
```diff
@@ -34,11 +34,12 @@ import {
   // @ts-ignore
   POLYGON_MAINNET_RPC,
 } from 'react-native-dotenv';
+import { setRpcEndpoints, web3SetHttpProvider } from '../handlers/web3';
```
Comment (estebanmino) [inline src/model/config.ts:37]: curious about the change of source here from `@rainbow-me/handlers/web3`

**Review by estebanmino — APPROVED** [review]
🚀 🚀 🐐 🐐

**Review by jinchung — APPROVED** [review]
🌮

**Review by brunobar79 — APPROVED** [review]
🚀

**Pull request merged**